frontend/src/metabase/querying/components/FilterModal/DateFilterEditor/DateFilterEditor.unit.spec.tsx

```diff
@@ -0,0 +1,215 @@
+    });
+
+    it("should add a relative date filter", async () => {
+      const { getNextFilterName } = setup({
+        query: defaultQuery,
+        stageIndex,
+        column,
+      });
+
+      userEvent.click(screen.getByLabelText("More options"));
+      userEvent.click(await screen.findByText("Last 30 days"));
+
+      expect(getNextFilterName()).toBe("Created At is in the previous 30 days");
+    });
+
+    it("should remove a relative date filter", async () => {
+      const { query, filter } = createQueryWithFilter(
+        defaultQuery,
+        stageIndex,
+        Lib.relativeDateFilterClause({
+          column,
+          value: -30,
+          bucket: "day",
+          offsetValue: null,
+          offsetBucket: null,
+          options: {},
+        }),
+      );
+      const { getNextFilterName } = setup({
+        query,
+        stageIndex,
+        column,
+        filter,
+      });
+      expect(screen.getByText("Previous 30 Days")).toBeInTheDocument();
+
+      userEvent.click(screen.getByLabelText("Clear"));
+      expect(getNextFilterName()).toBe(null);
+    });
+
+    it("should add a specific date filter", async () => {
+      const { getNextFilterName } = setup({
+        query: defaultQuery,
+        stageIndex,
+        column,
+      });
+
+      userEvent.click(screen.getByLabelText("More options"));
+      userEvent.click(await screen.findByText("Specific dates…"));
+      userEvent.click(screen.getByText("After"));
+      userEvent.clear(screen.getByLabelText("Date"));
+      userEvent.type(screen.getByLabelText("Date"), "Feb 15, 2020");
+      userEvent.click(screen.getByText("Add filter"));
+
+      expect(getNextFilterName()).toBe("Created At is after Feb 15, 2020");
+    });
+
+    it("should remove a specific date filter", async () => {
+      const { query, filter } = createQueryWithFilter(
+        defaultQuery,
+        stageIndex,
+        Lib.specificDateFilterClause(defaultQuery, stageIndex, {
+          operator: "=",
+          column,
+          values: [new Date(2020, 1, 15)],
+        }),
+      );
+      const { getNextFilterName } = setup({
+        query,
+        stageIndex,
+        column,
+        filter,
+      });
+      expect(screen.getByText("Feb 15, 2020")).toBeInTheDocument();
+
+      userEvent.click(screen.getByLabelText("Clear"));
+      expect(getNextFilterName()).toBe(null);
+    });
+
+    it("should add an exclude date filter", async () => {
+      const { getNextFilterName } = setup({
+        query: defaultQuery,
+        stageIndex,
+        column,
+      });
+
+      userEvent.click(screen.getByLabelText("More options"));
+      userEvent.click(await screen.findByText("Exclude…"));
+      userEvent.click(screen.getByText("Hours of the day…"));
+      userEvent.click(screen.getByText("5 PM"));
+      userEvent.click(screen.getByText("Add filter"));
+
+      expect(getNextFilterName()).toBe("Created At excludes the hour of 5 PM");
```

Comment:
the issue claims `screen.getByText("Created At excludes the hour of 5 PM") should be a different string.`

Reply:
yep. The idea is that this test should be updated when the issue is fixed.